code/modules/recycling/sortingmachinery.dm

```diff
@@ -93,6 +93,7 @@
 	if(href_list["remove_dest"] && mode)
 		var/idx = Clamp(text2num(href_list["remove_dest"]), 1, destinations.len)
 		destinations -= destinations[idx]
+		currTag = 0 // In case the index was at the end of the list
```

Comment:
Don't do this, it's bad behaviour and I bet we'll get a bug report for it. Just do
```
if(currTag == destinations[idx])
   currTag = 0
```
before the destination is removed.